### PR TITLE
feat: implement drag n' drop

### PR DESCRIPTION
**Pull request by jrandolf**

### CODE REVIEW

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 1]
```diff
@@ -641,6 +613,21 @@ export class ActionDispatcher {
         commands: command ? [command] : undefined,
       }
     );
+    // Drag cancelling happens on escape.
+    if (key === 'Escape') {
+      if (
+        !source.alt &&
+        ((this.#isMacOS && !source.ctrl && !source.meta) || !this.#isMacOS)
+      ) {
+        promise = Promise.all([
```
[the remark this answers is not in the document]
Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:622]: What do you mean? It gets returned at the end?

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 2]
```diff
@@ -624,7 +596,7 @@ export class ActionDispatcher {
         default:
```
[the remark this answers is not in the document]
Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:596]: The empty default is intentional.

[the remark this answers is not in the document]
Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:596]: I've left a note.

**Comment by OrKoN** [comment]
make sure https://github.com/GoogleChromeLabs/chromium-bidi/blob/main/.browser has the right version

File: src/bidiMapper/domains/input/ActionDispatcher.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thiagowfx) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:622]: Then I suppose the weird/confusing part is the empty `then`?

[the remark this answers is not in the document]
Comment (thiagowfx) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:622]: It's just not very clear what's happening here

[the remark this answers is not in the document]
Comment (Lightning00Blade) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:622]: I think we should just await the promise rather then returning it and having a then that would be easier to read imo.
As currently : Create Promise -> Wrap in All -> Add then to return undefine upon completion rather then [] -> Return -> Await else where
Proposition : Create Promise -> Wrap in all -> Await All -> return; (Empty)

**Review by thiagowfx — APPROVED** [review]

**Review by Lightning00Blade — APPROVED** [review]

**Review by OrKoN — APPROVED** [review]

**Pull request merged**